libs/feature/search/src/lib/favorites/favorite-star/favorite-star.component.spec.ts

```diff
@@ -150,19 +152,30 @@ describe('FavoriteStarComponent', () => {
           ])
           expect(favoritesService.addToFavorites).not.toHaveBeenCalled()
         })
+        beforeEach(() => {
```

Comment:
You should not add a `beforeEach` like that, always put it in a new `describe` section.
Note the different describe sections, they play a test scenario. As we changed the logic of the component, the scenario should be changed.
There are 2 potential actions
1. component toggle
2. `myFavorites$` tick, which can comes from another component.

Those 2 actions are not binded to the same toggle as before.
Here you are not testing the fact the `myFavorites$` could tick from another component, which wouldn't change the count of this component.